platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+#include "coding/file_reader.hpp"
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+
+/// @return directory where binary resides, including slash at the end
+static bool GetBinaryFolder(string & outPath)
```

Comment:
So basically, to get this PR accepted, you should make your changes in `platform_linux.cpp` instread.